pkg/sql/colexec/generate_series/generate_series.go

```diff
@@ -0,0 +1,282 @@
+	}
+	switch param.Cols[0].Typ.Id {
+	case int32(types.T_int32):
+		var start, end, step int64
+		start, err = strconv.ParseInt(param.start, 10, 32)
+		if err != nil {
+			return false, err
+		}
+		end, err = strconv.ParseInt(param.end, 10, 32)
+		if err != nil {
+			return false, err
+		}
+		if len(param.step) == 0 {
+			step = 1
+			if start > end {
+				step = -1
+			}
+		} else {
+			step, err = strconv.ParseInt(param.step, 10, 32)
+			if err != nil {
+				return false, err
+			}
+		}
+		res, err := generateInt32(int32(start), int32(end), int32(step))
+		if err != nil {
+			return false, err
+		}
+		for i := range res {
+			err = rbat.Vecs[0].Append(res[i], false, proc.Mp())
+			if err != nil {
+				return false, err
+			}
+		}
+		rbat.InitZsOne(len(res))
+	case int32(types.T_int64):
+		var start, end, step int64
+		start, err = strconv.ParseInt(param.start, 10, 64)
+		if err != nil {
+			return false, err
+		}
+		end, err = strconv.ParseInt(param.end, 10, 64)
+		if err != nil {
+			return false, err
+		}
+		if len(param.step) == 0 {
+			step = 1
+			if start > end {
+				step = -1
+			}
+		} else {
+			step, err = strconv.ParseInt(param.step, 10, 64)
+			if err != nil {
+				return false, err
+			}
+		}
+		res, err := generateInt64(start, end, step)
+		if err != nil {
+			return false, err
+		}
+		for i := range res {
+			err = rbat.Vecs[0].Append(res[i], false, proc.Mp())
+			if err != nil {
+				return false, err
+			}
+		}
+		rbat.InitZsOne(len(res))
+	case int32(types.T_datetime):
+		res, err := generateDatetime(param.start, param.end, param.step, param.Cols[0].Typ.Precision)
+		if err != nil {
+			return false, err
+		}
+		for i := range res {
+			err = rbat.Vecs[0].Append(res[i], false, proc.Mp())
+			if err != nil {
+				return false, err
+			}
+		}
+		rbat.InitZsOne(len(res))
+	}
+	proc.SetInputBatch(rbat)
+	return false, nil
+}
+
+func dupType(typ *plan.Type) types.Type {
+	return types.Type{
+		Oid:       types.T(typ.Id),
+		Width:     typ.Width,
+		Size:      typ.Size,
+		Precision: typ.Precision,
+	}
+}
+
+func judgeArgs[T Number](start, end, step T) ([]T, error) {
+	if step == 0 {
+		return nil, moerr.NewInvalidInput("step size cannot equal zero")
+	}
+	if start == end {
+		return []T{start}, nil
+	}
+	s1 := step > 0
+	s2 := end > start
+	if s1 != s2 {
+		return []T{}, nil
+	}
+	return nil, nil
+}
+
+func trimStep(step string) string {
+	step = strings.TrimSpace(step)
+	step = strings.TrimSuffix(step, "s")
+	step = strings.TrimSuffix(step, "(s)")
+	return step
+}
+
+func genStep(step string) (num int64, tp types.IntervalType, err error) {
+	step = trimStep(step)
+	s := strings.Split(step, " ")
+	if len(s) != 2 {
+		err = moerr.NewInvalidInput("invalid step '%s'", step)
+		return
+	}
+	num, err = strconv.ParseInt(s[0], 10, 64)
+	if err != nil {
+		err = moerr.NewInvalidInput("invalid step '%s'", step)
+		return
+	}
+	tp, err = types.IntervalTypeOf(s[1])
+	return
+}
+
+func generateInt32(start, end, step int32) ([]int32, error) {
```

Comment:
I'm not sure if it is suitable to do such check work (the ParseInt and so on above is also) at colexec operator. it's too heavy for me.
Maybe it is better to do that while we are compiling the Scope or make the plan.
let's see other reviewer says.